pkg/api/rest/resttest/resttest.go

```diff
@@ -667,10 +667,6 @@ func (t *Tester) testDeleteGracefulImmediate(obj runtime.Object, setFn SetFunc,
 	if !errors.IsNotFound(err) {
 		t.Errorf("unexpected error, object should be deleted immediately: %v", err)
 	}
-	objectMeta = t.getObjectMetaOrFail(out)
```

Comment:
I'd prefer you didn't remove this block, but continue to assert whatever behavior we enforce (in this case, the DeletionTimestamp is not set, the DeletionGracePeriodSeconds is not set).

Also, this may be a breaking change to clients that were relying on DeletionGracePeriodSeconds to be set - please check the Kubelet signals non-graceful deletion (should give it 2 seconds to delete, which is the minimum) instead of defaulting to the pod spec value.